### PR TITLE
Allow menu-trigger to take a menu interface.

### PR DESCRIPTION
**Pull request by trshafer**
This should enable menus to be more easily extended for custom menu implementations.

Addresses #1560.

### CODE REVIEW

**Comment by trshafer** [comment]
I went back and forth on whether or not to make it an abstract class with many of the implementation details on the abstract class. I went with the interface because it's less invasive to the consumer than an abstract class.

File: src/lib/menu/menu-interface.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {EventEmitter, TemplateRef} from '@angular/core';
+import {MenuPositionX, MenuPositionY} from './menu-positions';
+
+export interface MdMenuInterface {
+  positionX: MenuPositionX;
+  positionY: MenuPositionY;
+  templateRef: TemplateRef<any>;
+  close: EventEmitter<void>;
+  _focusFirstItem: () => void;
+  _setClickCatcher: (bool: boolean) => void;
```
Comment (jelbourn) [inline src/lib/menu/menu-interface.ts]: If these two are going to be part of this interface, they'll should lose the underscore (which implies "internal api only")

**Comment by trshafer** [comment]
Force pushed to address the conflict in merge of #1534.

File: src/lib/menu/menu-interface.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import {EventEmitter, TemplateRef} from '@angular/core';
+import {MenuPositionX, MenuPositionY} from './menu-positions';
+
+export interface MdMenuInterface {
```
Comment (jelbourn) [inline src/lib/menu/menu-interface.ts]: Let's call this `MdMenuPanel`
(generally trying to avoid using the word "interface" in the names of interfaces)

**Comment by trshafer** [comment]
Hey @jelbourn, I have made the requested updates. Good suggestions. The test really helps demonstrate the creation of custom menus.

**Comment by jelbourn** [comment]
LGTM aside from a couple last comments. @kara should also take a look

File: src/lib/menu/menu-panel.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import {EventEmitter, TemplateRef} from '@angular/core';
+import {MenuPositionX, MenuPositionY} from './menu-positions';
+
+export interface MdMenuPanel {
+  positionX: MenuPositionX;
+  positionY: MenuPositionY;
+  templateRef: TemplateRef<any>;
+  close: EventEmitter<void>;
+  _focusFirstItem: () => void;
```
Comment (kara) [inline src/lib/menu/menu-panel.ts]: Agree with @jelbourn that underscore should be removed if possible

**Comment by trshafer** [comment]
Hey @kara, thanks for the review. I have removed the underscore from the interface, rebased, squashed my commits, and updated the commit message style.

**Pull request merged**